### ISSUE TITLE
[FEA]: Random Number Generation

### ISSUE THREAD

**Issue by tomasruizt** [NVIDIA/cutile-python#17]
### Is this a new feature, an improvement, or a change to existing functionality?

New Feature

### How would you describe the priority of this feature request?

Critical (currently preventing usage)

### Please provide a clear description of problem this feature solves

I searched in the documentation for random number generation functions but have not found any so far. 

As a reference, see Triton's API: https://triton-lang.org/main/python-api/triton.language.html#random-number-generation

### Feature Description

One usage example is to create noise shaped with the size of the tiles. E.g for gumbel-max sampling.

### Describe your ideal solution

The Triton API is a good starting point.

### Contributing Guidelines

- [x] I agree to follow cuTile Python's contributing guidelines
- [x] I have searched the [open feature requests](https://github.com/nvidia/cutile-python/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request) and have found no duplicates for this feature request

**Comment by HiralThadeshwar31** [comment]
@tomasruizt Could you assign me this issue?
Thanks,
Hiral

**Comment by ZhangZhiPku** [comment]
> I don't own the repository and I don't have the authority to assign issues exclusively to people. [@HiralThadeshwar31](https://github.com/HiralThadeshwar31)

following code would serve as an alternative way:
```python
import torch
import cuda.tile as ct

@ct.function
def tile_rnd(seed: ct.Tile) -> ct.Tile:
    return seed * 1103515245 + 12345

@ct.kernel
def device_rnd(rnd: ct.Tile, tile_size: ct.Constant):
    block_idx = ct.bid(0)
    seed = ct.arange(tile_size, dtype=ct.int32) + block_idx * tile_size
    ct.store(rnd, (block_idx, ), tile_rnd(seed))


n, tile_size = 2048, 512
num_blocks = (n // tile_size, )
out = torch.empty(size=[n], device="cuda", dtype=torch.int)

ct.launch(
    torch.cuda.current_stream(), num_blocks, 
    device_rnd, (out, tile_size)
)

print(out)

```

**Comment by tomasruizt** [comment]
@haijieg now that Tile IR is announced, do you have a rough idea what building blocks / changes are needed to add the functions of the random API?
